optaplanner-examples/src/main/java/org/optaplanner/examples/common/swingui/SolverAndPersistenceFrame.java

```diff
@@ -170,51 +176,57 @@ private JComponent createQuickOpenPanel() {
     }
 
     private JComponent createQuickOpenUnsolvedPanel() {
-        quickOpenUnsolvedPanel = new JPanel();
+        DefaultListModel<QuickOpenAction> model = new DefaultListModel<>();
+        quickOpenUnsolvedPanel = new JList<>(model);
+        listModels.put(quickOpenUnsolvedPanel, model);
         quickOpenUnsolvedActionList = new ArrayList<>();
         List<File> unsolvedFileList = solutionBusiness.getUnsolvedFileList();
         return createQuickOpenPanel(quickOpenUnsolvedPanel, "Unsolved dataset", quickOpenUnsolvedActionList,
                 unsolvedFileList);
     }
 
     private JComponent createQuickOpenSolvedPanel() {
-        quickOpenSolvedPanel = new JPanel();
+        DefaultListModel<QuickOpenAction> model = new DefaultListModel<>();
+        quickOpenSolvedPanel = new JList<>(model);
+        listModels.put(quickOpenSolvedPanel, model);
         quickOpenSolvedActionList = new ArrayList<>();
         List<File> solvedFileList = solutionBusiness.getSolvedFileList();
         return createQuickOpenPanel(quickOpenSolvedPanel, "Solved dataset", quickOpenSolvedActionList,
                 solvedFileList);
     }
 
-    private JComponent createQuickOpenPanel(JPanel panel, String title, List<Action> quickOpenActionList, List<File> fileList) {
-        panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
-        refreshQuickOpenPanel(panel, quickOpenActionList, fileList);
-        JScrollPane scrollPane = new JScrollPane(panel);
+    private JComponent createQuickOpenPanel(JList<QuickOpenAction> listPanel, String title, List<Action> quickOpenActions, List<File> files) {
+        listPanel.setSelectionMode(SINGLE_SELECTION);
+        listPanel.addListSelectionListener(event -> {
+            QuickOpenAction action = listPanel.getModel().getElementAt(listPanel.getSelectedIndex());
+            action.actionPerformed(null);
```

Comment:
And there's another even more obscure bug in this code. When the model is cleared in `refreshQuickOpenPanel()` selection changes and `ListSelectionEvent` is fired (with `firstIndex=0`) but the `listPanel` has `slectedIndex=-1`. This causes `ArrayIndexOutOfBoundsException`. So another condition to check before performing the action.

Reply:
One more thing needs reworking and it's disabling data set loading during solving. Currently, when Solve action is performed, all load actions are disabled. This worked when the actions were associated with buttons because the buttons got disabled too. So now we need to disable the whole JListPanel and re-enable it once solving is terminated. On the other hand we could get rid of `quickOpenUnsolvedActionList` and `quickOpenSolvedActionList` since we will no longer need to disable/enable individual actions.